src/Microsoft.AspNet.Mvc.Core/ViewComponents/ViewComponentDescriptor.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Mvc.ViewComponents
+{
+    /// <summary>
+    /// A descriptor for a View Component.
+    /// </summary>
+    public class ViewComponentDescriptor
+    {
+        /// <summary>
+        /// Gets or sets the full name.
+        /// </summary>
+        public string FullName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the short name.
+        /// </summary>
+        public string ShortName { get; set; }
```

Comment:
should probably describe
- restrictions e.g. a requirement that `ShortName` be a substring of `FullName`
- uniqueness requirements: both `ShortName` and `FullName` are used to find view components. must both be unique within the application?